Assets/Scripts/KingAI.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class KingAI : MonoBehaviour
+{
+    public Transform player; 
+    private NavMeshAgent agent;
+    private Animator anim;
+
+    void Start()
+    {
+        agent = GetComponent<NavMeshAgent>();
+        anim = GetComponent<Animator>();
```

Comment:
make sure to call `anim.SetInteger("WeaponType_int", 0);` to stop him from punching by default. The animator controller isnt checked in to git so if you changed the default through the UI it won't be shared with us